cmd/cli/app/provider/provider_enroll.go

```diff
@@ -98,28 +109,56 @@ func EnrollProviderCommand(ctx context.Context, cmd *cobra.Command, _ []string,
 		}
 	}
 
+	config, err := providerConfigFromArg(cfgFlag, cmd.InOrStdin())
+	if err != nil {
+		return cli.MessageAndError("Error reading provider configuration", err)
+	}
+
 	// the token only applies to the old flow
+	// TODO: allow for token to be passed in if the provider allows it, don't hardcode
 	if token != "" && provider == legacyGitHubProvider.ToString() {
```

Comment:
this needs to be changed to either a switch/case of known providers or better to subcommands per provider.